feat: add colorpicker components

hzhu opened the pull request:
## Description

This PR adds the website documentation for `react-colorpickers`.

## Checklist

- [ ] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [ ] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

jzempel commented on src/pages/components/colorpickers.mdx:
```diff
@@ -0,0 +1,68 @@
+
+## How to use it
+
+### Color picker
```
Use `Default` for consistency.

steue replied:
good call, i've added some copy and an example to our drafts.

jzempel commented on src/pages/components/colorpickers.mdx:
```diff
@@ -0,0 +1,68 @@
+  <ColorpickerDefault />
+</CodeExample>
+
+### Color picker dialog
```
Use `Dialog` for alignment with other component pages where we rarely use the heading to refer back to the API component name.

jzempel reviewed (commented):
@steue a couple of comments while this is in flight. At a minimum, it feels like there should be another example demonstrating the use of a custom trigger. https://zendeskgarden.github.io/react-components/?path=/story/components-colorpickers-colorpickerdialog--with-icon-button

steue reviewed (commented):
looks good to me! thanks henry!

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+                }
+              }}
+            />
+            <ColorpickerDialog
```
Missing `focusInset` prop.

jzempel replied:
Also the button is missing an `aria-label` for accessibility.

jzempel commented on src/examples/colorpickers/ColorpickerDialogDefault.tsx:
```diff
@@ -0,0 +1,20 @@
+
+const Example = () => (
+  <Row justifyContent="center">
+    <Col size="auto">
```
Replace `size` with `textAlign="center"`.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+
+  return (
+    <Row justifyContent="center">
+      <Col size="auto">
```
Much of the follow-on `StyledXxx` customization for positioning can be alleviated by replacing this with a `StyledCol` that sets `flex-grow: 0;`.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+  return (
+    <Row justifyContent="center">
+      <Col size="auto">
+        <StyledField>
```
Use `Field`.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+      <Col size="auto">
+        <StyledField>
+          <Label>Favorite color</Label>
+          <StyledInputGroup>
```
Use `InputGroup`.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+          <Label>Favorite color</Label>
+          <StyledInputGroup>
+            <StyledInput
+              isCompact
```
Remove unused `isCompact`.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+
+const Example = () => {
+  const [input, setInput] = useState('#1f73b799');
+  const [color, setColor] = useState<string | IColor>('rgba(31, 115, 183, .6)');
```
Use `PALETTE.blue[600]` together with the `rgba` function from polished.js.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+};
+
+const Example = () => {
+  const [input, setInput] = useState('#1f73b799');
```
Use `PALETTE.blue[600]` with `99` string interpolation.

jzempel commented on src/examples/colorpickers/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,91 @@
+`;
+
+const StyledInput = styled(Input)`
+  max-width: ${props => props.theme.space.base * 26}px;
```
Let's apply a base-4 max-width that accommodates the widest possible value, which I believe would be `#AAAAAAAA`. I think `base * 27` should do it.

steue commented on src/nav/components.yml:
```diff
@@ -48,6 +48,8 @@
           title: Toggle icon button
     - title: Chrome
       id: /components/chrome
+    - title: Colorpickers
+      id: /components/colorpickers
```
Thanks for the clarification jz! I noticed this as well, but thought this was intentional.

steue commented on src/pages/components/colorpickers.mdx:
```diff
@@ -0,0 +1,90 @@
+
+### Dialog
+
+The Color picker dialog allows users to use a color picker within a dialog.
```
...allows users to operate? a color picker within a dialog.

allisonacs replied:
How about

> The Color picker can be placed inside a dialog.

steue replied:
Ah yeah that's so much more simple.

jzempel commented on src/examples/color-picker/ColorpickerDialogDefault.tsx:
```diff
@@ -0,0 +1,24 @@
+  <Row>
+    <Col textAlign="center">
+      <ColorpickerDialog
+        aria-label="Color picker dialog button"
```
[nit] maybe something closer to reality, like "Select your favorite color"

But more important, did you test this? The props are spread on to the `Colorpicker` div – as expected. But this might identify a short-coming in the API. Perhaps the need for `triggerProps` (or something better?) on the underlying component. Otherwise, VO reads "button".

hzhu replied:
I hadn't noticed that props are not passed through to the trigger. There is a new PR (https://github.com/zendeskgarden/react-components/pull/1071) in draft mode that allows consumers to apply props to the trigger.

hzhu replied:
The `aria-label` has been removed from this PR. I'll open a new PR to add the `aria-label` via `buttonProps` once  https://github.com/zendeskgarden/react-components/pull/1071 is released.

jzempel commented on src/examples/color-picker/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,95 @@
+const StyledInput = styled(Input)`
+  max-width: ${props => props.theme.space.base * 27}px;
+`;
+const toHex = (selectedColor: IColor) => {
```
[nit] add line break above.

jzempel commented on src/examples/color-picker/ColorpickerDialogFormInput.tsx:
```diff
@@ -0,0 +1,95 @@
+            <ColorpickerDialog
+              focusInset
+              color={color}
+              aria-label="Color picker dialog button"
```
[nit] improve label content. But per previous comment, no VO indication of what this button is for.

jzempel reviewed (approved):
why is the `react-components` submodule updated here? Shouldn't it remain at the v8.35.0 tag?

jzempel commented on src/examples/color-picker/ColorpickerDialogCustomTrigger.tsx:
```diff
@@ -0,0 +1,51 @@
+  ) => (
+    <Tooltip content="Palette">
+      <IconButton aria-label="palette" ref={ref} {...props}>
+        <PaletteIcon style={{ color: props.iconColor }} />
```
The alpha value doesn't appear to be getting applied to this icon.

hzhu replied:
Looks like the alpha changes broke using the hex color from `DEFAULT_THEME`.  The value is now passed to `rgba` and alpha should now be applied to the icon.

hzhu commented:
> why is the react-components submodule updated here? Shouldn't it remain at the v8.35.0 tag?

I wasn't aware this was checked in. I've removed the change and rebased the commits.
